kratos/geometries/geometry.h

```diff
@@ -731,6 +731,54 @@ class Geometry
     ///@name Inquiry
     ///@{
 
+    /**
+     * @brief Checks if two GeometryType have the same type
+     * @return True if the objects are the same type, false otherwise
+     */
+    inline static bool HasSameType(const GeometryType& rLHS, const GeometryType& rRHS) {
+        return (typeid(rLHS) == typeid(rRHS));
+    }
+
+    /**
+     * @brief Checks if two GeometryType have the same type (pointer version)
+     * @return True if the objects are the same type, false otherwise
+     */
+    inline static bool HasSameType(const GeometryType * rLHS, const GeometryType* rRHS) {
+        return GeometryType::HasSameType(*rLHS, *rRHS);
+    }
+
+    /**
+     * @brief Checks if two GeometryType have the same geometry type
+     * @return True if the geometries are the same type, false otherwise
+     */
+    inline static bool HasSameGeometryType(const GeometryType& rLHS, const GeometryType& rRHS) {
+        return (rLHS.GetGeometryType() == rRHS.GetGeometryType());
+    }
+
+    /**
+     * @brief Checks if two GeometryType have the same geometry type (pointer version)
+     * @return True if the geometries are the same type, false otherwise
+     */
+    inline static bool HasSameGeometryType(const GeometryType* rLHS, const GeometryType* rRHS) {
+        return GeometryType::HasSameGeometryType(*rLHS, *rRHS);
+    }
+
+    /**
+     * @brief Checks if two GeometryType are the same
+     * @return True if the object is the same, false otherwise
+     */
+    inline static bool IsSame(const GeometryType& rLHS, const GeometryType& rRHS) {
```

Comment:
I would say using the same naming would be probably good. So it should probably start with `Has...` Or we rename the other functions to `Is...`

In C# this function would be name `IsEqual` maybe that would be the case, here?

Does this really ensure that it is exactly the same? Do you even need both comparisons isn't HasSameType already insisting that it is the same?

Reply:
I would use the same names as in the `GeometricalObject`

Reply:
ok, then the names are correct as they are.